src/repr/src/row.rs

```diff
@@ -2031,6 +2031,29 @@ impl RowPacker<'_> {
     where
         I: IntoIterator<Item = D>,
         D: Borrow<Datum<'a>>,
+    {
+        self.push_array_with(dims, |packer| {
+            let mut nelements = 0;
+            for datum in iter {
+                packer.push(datum);
+                nelements += 1;
+            }
+            Ok::<_, InvalidArrayError>(nelements)
+        })
+    }
+
+    /// Convenience function to construct an array from a function. The function must return the
+    /// number of elements it pushed into the array. It is undefined behavior if the function returns
+    /// a number different to the number of elements it pushed.
```

Comment:
@ParkMyCar may know this better than me, but if a function can exhibit undefined behavior, isn't it best practice to mark it as `unsafe`?

Reply:
Yeah slight preference to either marking this function as unsafe and/or renaming it to `push_array_with_unchecked(...)` and renaming `push_array(...)` to `try_push_array(...)`.